jwst/pipeline/calwebb_image3.py

```diff
@@ -100,6 +100,7 @@ def process(self, input):
         self.save_model(result, suffix=self.suffix)
 
         self.log.info("Creating source catalog...")
+        self.source_catalog.save_results = self.save_results
```

Comment:
So in all of these cases that now pass `self.save_results` (which is the pipeline-level argument) to the lower-level step arguments, where is the pipeline-level argument being set? Is that still in the pipeline cfg file? If so, then what about cases where the pipeline cfg isn't being used?

Reply:
The pipeline level `save_results` is handled as before: either set in `strun` or, if a user is doing so programmatically, set explicitly. No change in behavior there.

The intent of this PR is that, for all pipelines, if the "expected results" of that pipeline come from more than one step, those steps are forced to follow the pipeline's state of `save_results`.